model/optimize/utils.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package optimize
+
+import (
+	"bytes"
+
+	"github.com/unidoc/unipdf/v3/core"
+)
+
+type objectStructure struct {
+	catalogDict *core.PdfObjectDictionary
+	pagesDict   *core.PdfObjectDictionary
+	pages       []*core.PdfIndirectObject
+}
+
+// getObjectStructure identifies the Catalog and Pages dictionary and finds a list of pages.
+func getObjectStructure(objects []core.PdfObject) objectStructure {
+	objstr := objectStructure{}
+	found := false
+	for _, obj := range objects {
+		switch t := obj.(type) {
+		case *core.PdfIndirectObject:
+			dict, is := core.GetDict(t)
+			if !is {
+				continue
+			}
+			kind, is := core.GetName(dict.Get("Type"))
+			if !is {
+				continue
+			}
+
+			switch kind.String() {
+			case "Catalog":
+				objstr.catalogDict = dict
+				found = true
+			}
+		}
+		if found {
+			break
+		}
+	}
+
+	if !found {
+		return objstr
+	}
+
+	pagesDict, ok := core.GetDict(objstr.catalogDict.Get("Pages"))
+	if !ok {
+		return objstr
+	}
+	objstr.pagesDict = pagesDict
+
+	kids, ok := core.GetArray(pagesDict.Get("Kids"))
+	if !ok {
+		return objstr
+	}
+	for _, obj := range kids.Elements() {
+		pobj, ok := core.GetIndirect(obj)
+		if !ok {
+			break
```

Comment:
Should this be `continue` here instead of `break`?

Reply:
This is something that should not happen, I think.  Might make sense to log a debug message.  It's part of the optimization so it's not very mission critical.